### PR TITLE
Refactored data list

### PR DESCRIPTION
**Pull request by SUdoWinchester**

### CODE REVIEW

File: project/resources/js/Components/DataList.vue [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
-
-    props: {
+      components: {Pagination},
+      props: {
       dataSource: {
         type: String,
       },
     },
 
-    watch: {
```
Comment (danilopinotti) [inline project/resources/js/Components/DataList.vue:13]: Why did you remove this?

File: project/resources/js/Components/DataList.vue [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
+            <tbody>
+                <slot name="body" :fetchData="fetchData" :items="items"></slot>
+            </tbody>
+        </table>
+
+        <pagination
+          class="col-10"
+          :total-pages="totalPages"
+          :current-page="currentPage"
+          @fetchPrevPage="fetchPrevPage"
```
Comment (danilopinotti) [inline project/resources/js/Components/DataList.vue:19]: `@fetchNextPage` and `@fetchPrevPage` may be omitted when using only `@changePage`

File: project/resources/js/Components/DataListFilters/Text.vue [hunk 3]
```diff
@@ -11,6 +11,13 @@ export default {
 
     mixins: [filter],
 
+    props: {
+        ariaPlaceholder: {
```
Comment (danilopinotti) [inline project/resources/js/Components/DataListFilters/Text.vue:15]: `placeholder`

File: project/resources/js/Components/Pagination.vue [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    },
+    currentPage: {
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      paginationButtons: [],
+      currentPageData: 1,
```
Comment (danilopinotti) [inline project/resources/js/Components/Pagination.vue:53]: `currentPageMutable`
CurrentPageData sounds like 'the data contained inside the current page'

File: project/resources/js/Components/DataList.vue (hunk 1 above)
Comment (SUdoWinchester) [reply project/resources/js/Components/DataList.vue:13]: The text filter is in the text component. Debounce is more appropriate there.
